### PR TITLE
Add maximum_grade calculated attribute

### PR DESCRIPTION
**Pull request by minhtule**

### CODE REVIEW

**Comment by minhtule** [comment]
@joel can give me some hint how to reuse the calculated attribute in the scope?

**Comment by lowjoel** [comment]
...just use it like normal? scopes are just like normal class methods

**Comment by minhtule** [comment]
Wow, -0.00003% decrease...

**Comment by minhtule** [comment]
The file I changed is [100% covered](https://coveralls.io/builds/4438481/source?filename=app%2Fmodels%2Fcourse%2Fassessment.rb). Why does the coverage drop? >.< @lowjoel  help please!

**Comment by minhtule** [comment]
I think the coverage drops because I deleted more lines than I added.

File: app/models/course/assessment.rb [hunk 1]
```diff
@@ -26,6 +26,16 @@ class Course::Assessment < ActiveRecord::Base
...
   has_many :submissions, inverse_of: :assessment, dependent: :destroy
 
+  # @!attribute [r] maximum_grade
+  #   Gets the maximum grade allowed by this assessment. This is the sum of all questions'
+  #   maximum grade.
+  #   @return [Fixnum]
+  calculated :maximum_grade, (lambda do
+    Course::Assessment::Question.where do
+      course_assessment_questions.assessment_id == course_assessments.id
+    end.select { sum(course_assessment_questions.maximum_grade) }
```
Comment (lowjoel) [inline app/models/course/assessment.rb]: move the sum to the topc.

**Comment by minhtule** [comment]
@lowjoel Only 2 lines now so the coverage drops even more. Please review! :grin:

**Pull request merged**